doc/source/configuration/ci-cd.rst

```diff
@@ -0,0 +1,179 @@
+
+    # Dictionary of runners to be deployed using the role.
+    # Each dict item can be provided with optional attributes
+    # * labels - provide a list of labels for a specific runner
+    #   overriding the contents of `default_runner_labels`
+    # * state - either `started`` or `absent`. By default it will
+    #   be started if however the runner needs to be removed
+    #   then setting it to `absent` will unregister the runner with
+    #   GitHub and remove it from the system.
+    # Example
+    # github_runners:
+    #  runner_01: {}
+    #  runner_02:
+    #    labels: ['foo', 'bar', 'baz']
+    #  runner_03:
+    #    state: absent
+    github_runners:
+      runner_01: {}
+      runner_02: {}
+      runner_03: {}
+
+    docker_users:
+      - "{{ runner_user }}"
+
+    pip_install_packages:
+      - name: docker
+
+If using multiple environments add an extra label to :code:`default_runner_labels` to distinguish these runners from runners belonging to other environments.
+Also feel free to change the number of runners and their names.
+
+6. Obtain a personal access token that would enable the registration of GitHub runners against the `github_account` and `github_repo` defined above.
+    This token ideally should be `fine-grained personal access token <https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token>`__ which may require the organisation to enable such tokens beforehand.
+    Steps can be found `here <https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization>`__.
+    The repository permissions for a fine-grained personal access token should be; :code:`Actions: R/W, Administration: R/W, Metadata: R`
+    Once the key has been obtained, add it to :code:`secrets.yml` under :code:`secrets_github_access_token`
+
+7. If the host is an actual Infra VM then please refer to upstream `Infrastructure VMs <https://docs.openstack.org/kayobe/latest/configuration/reference/infra-vms.html>`__ documentation for additional configuration and steps.
+
+8. Run :code:`kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/deploy-github-runner.yml`
+
+9. Check runners have registered properly by visiting the repository's :code:`Action` tab -> :code:`Runners` -> :code:`Self-hosted runners`
+
+10. Repeat the above steps for each environment you intend to deploy runners within.
+    You can share the fine-grained access token between environments.
+
+Workflow Deployment
+-------------------
+
+1. Edit `${KAYOBE_CONFIG_PATH}/inventory/group_vars/github-writer/writer.yml` in the base configuration making the appropriate changes to your deployments specific needs. See documentation for `stackhpc.kayobe_workflows.github <https://github.com/stackhpc/ansible-collection-kayobe-workflows/tree/main/roles/github>`__.
```

Comment:
This file doesn't exist any more. Could create a stub?

Reply:
Added.